spring-cloud-gcp-data-firestore/src/main/java/com/google/cloud/spring/data/firestore/FirestoreTemplate.java

```diff
@@ -62,61 +65,75 @@
  * @since 1.2
  */
 public class FirestoreTemplate implements FirestoreReactiveOperations {
-	private FirestoreClassMapper classMapper;
 
 	private static final int FIRESTORE_WRITE_MAX_SIZE = 500;
 
-	private static final String NAME_FIELD = "__name__";
+	private static final String NOT_FOUND_DOCUMENT_MESSAGE = "NOT_FOUND: Document";
 
-	private static final StructuredQuery.Projection ID_PROJECTION = StructuredQuery.Projection.newBuilder()
-			.addFields(StructuredQuery.FieldReference.newBuilder().setFieldPath(NAME_FIELD).build())
-			.build();
+	private final String nameField = FieldPath.documentId().toString();
 
-	private static final DocumentMask NAME_ONLY_MASK = DocumentMask.newBuilder().addFieldPaths(NAME_FIELD).build();
+	private final StructuredQuery.Projection idProjection = StructuredQuery.Projection.newBuilder()
+			.addFields(StructuredQuery.FieldReference.newBuilder().setFieldPath(nameField).build())
+			.build();
 
-	private static final String NOT_FOUND_DOCUMENT_MESSAGE = "NOT_FOUND: Document";
+	private final DocumentMask nameOnlyMask = DocumentMask.newBuilder().addFieldPaths(nameField).build();
 
-	private final FirestoreStub firestore;
+	private final FirestoreStub firestoreStub;
 
 	private final String parent;
 
 	private final String databasePath;
 
 	private final FirestoreMappingContext mappingContext;
 
+	private final Firestore firestore;
```

Comment:
Why do we need both `Firestore` and `FirestoreStub`?

Reply:
This was one of the experiments we did to determine feasibility. I think we can do without the extra client library object (`Firestore`) now.
`FirestoreStub` is used because the template implementation historically bypasses the client library and instead relies directly on gRPC layer. After our experiences with Spanner, I'd much rather rely on client library -- something to consider for the future, but far out of scope for this PR.

Reply:
Thanks for catching this. Sorry, I had left some experimental code in. After applying the changes @elefeint described [here](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/506#pullrequestreview-687777943), we no longer need to use the `Firestore` object.